add-on/content/SerpMonitor.jsm

```diff
@@ -96,6 +99,20 @@ this.SerpMonitor = {
   // tabs where the last visited page was a SERP.
   _serpTabs: new Map(),
 
+  get distributionId() {
+    if (!this._distributionId) {
+      this._distributionId = Cc["@mozilla.org/xre/app-info;1"].getService(Ci.nsIXULRuntime).distributionId;
+      if (!this._distributionId) {
+        this._distributionId = "";
+      }
+    }
+    return this._distributionId;
+  },
+
+  init(guid) {
+    this._guid = guid;
```

Comment:
Perf nit: Spidermonkey engineers have advised us in the past to avoid using expandos, because it makes the JS object change Shape and take a slow path using dictionary lookups for every property access. In this case we would declare _guid and _distributionId inside the SerpMonitor declaration initialized to null.

Reply:
Thanks, I've included this in the update for the ping data.